platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java

```diff
@@ -126,31 +128,38 @@ URL urlFromPath(String location) {
      * @return Properties from the file
      */
     Properties loadProperties(String location) {
-        return PropertiesLoader.loadProperties(location);
+        return PropertiesLoader.getInstance()
+                .loadProperties(location);
     }
 
     /**
-     * Runnable to refresh the CRL from the URL when either the nextUpdate time has elasped
+     * Runnable to refresh the CRL from the URL when either the nextUpdate time has elapsed
      * or it has rolled over to the next day.
      */
     private class CrlRefresh implements Runnable {
+        private Lock lock = new ReentrantLock();
 
         private Calendar start = Calendar.getInstance();
 
         public void run() {
-            Properties encryptionProperties = loadProperties(encryptionPropertiesLocation);
-            CRL crl = crlCache.get();
-            if (crl instanceof X509CRL) {
-                X509CRL x509Crl = (X509CRL) crl;
-                if (Calendar.getInstance()
-                        .getTime()
-                        .after(x509Crl.getNextUpdate()) || rolledOverDay()) {
-                    setCrlLocation(encryptionProperties.getProperty(CRL_PROPERTY_KEY));
-                    start = Calendar.getInstance();
+            String crlLocation = loadProperties(encryptionPropertiesLocation).getProperty(
+                    CRL_PROPERTY_KEY);
+
+            lock.lock();
+            try {
+                CRL crl = crlCache.get();
+                if (crl instanceof X509CRL) {
+                    X509CRL x509Crl = (X509CRL) crl;
+                    if (Calendar.getInstance()
+                            .getTime()
+                            .after(x509Crl.getNextUpdate()) || rolledOverDay()) {
```

Comment:
That would be wrong, as it no longer sets the location when it wasn't previously set. I've updated it to handle the NPE on `getNextUpdate()`.